apps/antalmanac/src/components/Header/LoadSaveFunctionality.tsx

```diff
@@ -32,6 +33,17 @@ interface LoadSaveButtonBaseState {
     rememberMe: boolean;
 }
 
+class SaveLoadIcon extends PureComponent<{ loading: boolean; actionName: 'Save' | 'Load' }> {
+    render() {
+        const { loading, actionName } = this.props;
+        if (loading) {
+            return <CircularProgress size={20} color={'inherit'} />;
+        } else {
+            return actionName === 'Save' ? <Save /> : <CloudDownload />;
+        }
+    }
+}
+
```

Comment:
Is there a reason you're using a legacy class-based component instead of a functional one? Prefer functional components, and leave a comment documenting why it needs to be a class component if that's the case (i.e. maybe the button base is a class component that's incompatible with function components as its start icon?)

Reply:
no, there is no specific reason. I just want to maintain consistency like how `LoadSaveButtonBase` class did that already in the file. I can change it to the functional components if that's preferred in React.